contracts/ethregistrar/BulkRenewal.sol

```diff
@@ -3,46 +3,73 @@ pragma experimental ABIEncoderV2;
 
 import "../registry/ENS.sol";
 import "./ETHRegistrarController.sol";
+import "./IETHRegistrarController.sol";
 import "../resolvers/Resolver.sol";
+import "./IBulkRenewal.sol";
 
-contract BulkRenewal {
-    bytes32 constant private ETH_NAMEHASH = 0x93cdeb708b7545dc668eb9280176169d1c33cfd8ed6f04690a0bcc88a93fc4ae;
-    bytes4 constant private REGISTRAR_CONTROLLER_ID = 0x018fac06;
-    bytes4 constant private INTERFACE_META_ID = bytes4(keccak256("supportsInterface(bytes4)"));
-    bytes4 constant public BULK_RENEWAL_ID = bytes4(
-        keccak256("rentPrice(string[],uint)") ^
-        keccak256("renewAll(string[],uint")
-    );
+import "@openzeppelin/contracts/utils/introspection/IERC165.sol";
 
-    ENS public ens;
+contract BulkRenewal is IBulkRenewal {
+    bytes32 private constant ETH_NAMEHASH =
+        0x93cdeb708b7545dc668eb9280176169d1c33cfd8ed6f04690a0bcc88a93fc4ae;
 
-    constructor(ENS _ens) public {
+    ENS public immutable ens;
+
+    constructor(ENS _ens) {
         ens = _ens;
     }
 
-    function getController() internal view returns(ETHRegistrarController) {
+    function getController() internal view returns (ETHRegistrarController) {
         Resolver r = Resolver(ens.resolver(ETH_NAMEHASH));
-        return ETHRegistrarController(r.interfaceImplementer(ETH_NAMEHASH, REGISTRAR_CONTROLLER_ID));
+        return
+            ETHRegistrarController(
+                r.interfaceImplementer(
+                    ETH_NAMEHASH,
+                    type(IETHRegistrarController).interfaceId
+                )
+            );
     }
 
-    function rentPrice(string[] calldata names, uint duration) external view returns(uint total) {
+    function rentPrice(string[] calldata names, uint256 duration)
+        external
+        view
+        override
+        returns (uint256 total)
+    {
         ETHRegistrarController controller = getController();
-        for(uint i = 0; i < names.length; i++) {
-            total += controller.rentPrice(names[i], duration);
+        for (uint256 i = 0; i < names.length; i++) {
+            (uint256 basePrice, uint256 premium) = controller.rentPrice(
+                names[i],
+                duration
+            );
+            total += (basePrice + premium);
         }
     }
 
-    function renewAll(string[] calldata names, uint duration) external payable {
+    function renewAll(string[] calldata names, uint256 duration)
+        external
+        payable
+        override
+    {
         ETHRegistrarController controller = getController();
-        for(uint i = 0; i < names.length; i++) {
-            uint cost = controller.rentPrice(names[i], duration);
-            controller.renew{value:cost}(names[i], duration);
+        for (uint256 i = 0; i < names.length; i++) {
+            (uint256 basePrice, uint256 premium) = controller.rentPrice(
+                names[i],
+                duration
+            );
+            controller.renew{value: basePrice + premium}(names[i]);
```

Comment:
People should be paying premium only at registration, not for renewal.  Maybe add `require(premium == 0)` instead?